vault/identity_store.go

```diff
@@ -239,6 +239,10 @@ func (i *IdentityStore) Invalidate(ctx context.Context, key string) {
 	}
 }
 
+func (i *IdentityStore) DB() *memdb.MemDB {
```

Comment:
I am concerned about exposing the raw memdb instance publicly on the identity store.  One technique I've seen to address this is to put this glue code in a _test.go file which will only expose it for tests.

Reply:
If we want to avoid exposing the memdb instance, we'll have to come up with a different mechanism. The test is in a package `vault_test` and not `vault`. So even for the test, this glue code will be public access.

Reply:
I tried it and put it in vault_store_test.go and it worked for me.

Reply:
I think its a fine case for `UpsertGroup` function to exist. I'll just revert deletion of that method and that should do it.